lib/Extractor/KLEEBuilder.cpp

```diff
@@ -136,7 +140,24 @@ ref<Expr> ExprBuilder::makeSizedArrayRead(unsigned Width, StringRef Name,
       new Array(ArrayNames.makeName(NameStr), 1, 0, 0, Expr::Int32, Width));
   ArrayVars.push_back(Origin);
 
+  std::vector<unsigned> ZeroBits, OneBits;
   UpdateList UL(Arrays.back().get(), 0);
+  if (Origin && Origin->K == Inst::Var && (Origin->KnownZeros != APInt(Width, 0) ||
+                                           Origin->KnownOnes != APInt(Width, 0))) {
+    ref<Expr> KnownZerosExpr = klee::ConstantExpr::create(0, Expr::Bool);
+    ref<Expr> KnownOnesExpr = klee::ConstantExpr::create(0, Expr::Bool);
+    ref<Expr> Var = ReadExpr::create(UL, klee::ConstantExpr::alloc(0, Expr::Int32));
+    // KnownZeros
+    ref<Expr> NotZeros = NotExpr::create(klee::ConstantExpr::alloc(Origin->KnownZeros));
+    ref<Expr> VarOrNotZero = OrExpr::create(Var, NotZeros);
```

Comment:
Here, in Var expression, the second argument is not the width of var, its an index value for a node to be added in updatelist UL. A linked list of var nodes is maintained and every time a variable is created, its added at the index of the list, that's why the index value is : 0 and index width is : Int32.